internal/Yesod/Session/Persist/Cookie/Logic.hs

```diff
@@ -0,0 +1,54 @@
+module Yesod.Session.Persist.Cookie.Logic
+  ( setCookie
+  , CookieContext (..)
+  ) where
+
+import Yesod.Session.Persist.Prelude
+
+import Yesod.Session.Persist.Cookie.SetCookie
+import Yesod.Session.Persist.Options
+import Yesod.Session.Persist.Save
+import Yesod.Session.Persist.Session
+
+import Yesod.Core.Types (Header)
+
+data CookieContext = CookieContext
+  { cookie :: Maybe ByteString
+  , load :: Maybe Session
+  , save :: Save Session
+  }
+
+setCookie :: Options tx m -> CookieContext -> [Header]
+setCookie options x =
+  let
+    cookiesForSession :: Maybe Session -> [Header]
+    cookiesForSession = makeSetCookieHeaders options . fmap (\s -> (s.key, s.time))
```

Comment:
Wow, where is record dot syntax coming from here? Enabled globally? I guess we'll see that soon if so.

Reply:
I don't have consensus on this, but it seemed harmless enough in this package...